### PR TITLE
Fix restricted mode when installing in non-canonical data dir

### PR DESCRIPTION
**Pull request by shlevy**

### CODE REVIEW

**Comment by shlevy** [comment]
Ping

**Comment by edolstra** [comment]
Why does it matter whether the install location is non-canonical? (What's the canonical location anyway?)

**Comment by shlevy** [comment]
Canonical as-in `canonPath`

**Comment by shlevy** [comment]
It matters because NIX_PATH lookup canonicalizes the path first, so it's disallowed in restricted mode.

**Comment by LnL7** [comment]
I think there's a similar problem when `$(pwd)` is a symlink, or is this related?

**Comment by shlevy** [comment]
This is related, yeah.

**Comment by shlevy** [comment]
@edolstra ping

File: src/libexpr/eval.cc [hunk 1]
```diff
@@ -311,7 +311,7 @@ EvalState::EvalState(const Strings & _searchPath, ref<Store> store)
     addToSearchPath("nix=" + settings.nixDataDir + "/nix/corepkgs");
 
     if (settings.restrictEval || settings.pureEval) {
-        allowedPaths = PathSet();
+        allowedPaths = {canonPath(settings.nixDataDir + "/nix/corepkgs", true)};
```
Comment (edolstra) [inline src/libexpr/eval.cc]: Isn't the correct fix to do
```
addToSearchPath("nix=" + canonPath(settings.nixDataDir + "/nix/corepkgs"));
 ```
?

Comment (shlevy) [reply src/libexpr/eval.cc]: Sure, I'll switch to that.

**Comment by shlevy** [comment]
@edolstra Fixed

**Pull request merged**